codeserver/ubi9-python-3.9/run-code-server.sh

```diff
@@ -13,60 +13,69 @@ if [ ! -f "/opt/app-root/src/.bashrc" ]; then
   echo 'PS1="\[\033[34;1m\][\$(pwd)]\[\033[0m\]\n\[\033[1;0m\]$ \[\033[0m\]"' > /opt/app-root/src/.bashrc
 fi
 
-# Initilize access logs for culling
+# Initialize access logs for culling
 echo '[{"id":"code-server","name":"code-server","last_activity":"'$(date -Iseconds)'","execution_state":"running","connections":1}]' > /var/log/nginx/codeserver.access.log
 
-# Add "/opt/app-root/src/.vscode/" directory to set default interpreter also for Run & Debug
-user_dir="/opt/app-root/src/.vscode/"
-settings_filepath="${user_dir}settings.json"
-launch_filepath="${user_dir}launch.json"
+# Function to create directories and files if they do not exist
+create_dir_and_file() {
+  local dir=$1
+  local filepath=$2
+  local content=$3
 
-json_launch_settings='{
-    // Use IntelliSense to learn about possible attributes.
-    // Hover to view descriptions of existing attributes.
-    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
-    "version": "0.2.0",
-    "configurations": [
-        {
-            "name": "Python Debugger: Current File",
-            "type": "debugpy",
-            "request": "launch",
-            "program": "${file}",
-            "console": "integratedTerminal",
-            "python": "/opt/app-root/bin/python3"
-        }
-    ]
+  if [ ! -d "$dir" ]; then
+    echo "Debug: Directory not found, creating '$dir'..."
+    mkdir -p "$dir"
+    echo "$content" > "$filepath"
+    echo "Debug: '$filepath' file created."
+  else
+    echo "Debug: Directory already exists."
+    if [ ! -f "$filepath" ]; then
+      echo "Debug: '$filepath' file not found, creating..."
+      echo "$content" > "$filepath"
+      echo "Debug: '$filepath' file created."
+    else
+      echo "Debug: '$filepath' file already exists."
+    fi
+  fi
+}
+
+# Define universal settings
+universal_dir="/opt/app-root/src/.local/share/code-server/User/"
+user_settings_filepath="${universal_dir}settings.json"
+universal_json_settings='{
+  "python.defaultInterpreterPath": "/opt/app-root/bin/python3",
+  "telemetry.telemetryLevel": "off",
+  "telemetry.enableTelemetry": false,
+  "telemetry.enableCrashReporter": false,
+  "telemetry.enableErrorTelemetry": false,
+  "workbench.enableExperiments": false
```

Comment:
Checking; so far, I see that Jupyter experiments may not be disabled, according to the screenshot below, because that's a separate setting, separate from https://github.com/atheo89/notebooks/commit/0d1ab4d33feb187e747f626e2d18b881ddc0a606#r143328849

![image](https://github.com/opendatahub-io/notebooks/assets/442720/031b1d8c-7d16-4369-b67b-887029e2d6ab)

Jupyter extension docs say that opting out of telemetry opts out of AB testing. Does that mean that experiments are disabled? https://github.com/microsoft/vscode-jupyter/wiki/Experiments

It would make sense to disable all experiments, IMO.

Reply:
Anyways, we can do this in a separate issue later.

Reply:
I checked again about this, as far as i understand since we set `workbench.enableExperiments: false` to the settings file it disables the participation in A/B testing and experimental features. This setting stops the editor from enabling experimental features that are being tested with a subset of users. So, I believe we are good  until proven otherwise!